x-pack/plugins/apm/server/lib/settings/agent_configuration/create_agent_config_index.ts

```diff
@@ -36,10 +36,11 @@ const mappings: Mappings = {
   dynamic_templates: [
     {
       // force string to keyword (instead of default of text + keyword)
+      // @ts-expect-error doesn't satisfy MappingDynamicTemplate
```

Comment:
looks like `type` is not part of `MappingDynamicTemplate`, and it should be no?

Reply:
It shouldn't be, but I'm not comfortable removing an unfamiliar piece of code.

Reply:
`type` is listed in one of the examples here: https://www.elastic.co/guide/en/elasticsearch/reference/current/dynamic-templates.html.

Reply:
updated the error, informed the upstream library